### PR TITLE
Migrate ILI9340 display driver to new logging subsystem

### PR DESCRIPTION
**Pull request by vanwinkeljan**

### CODE REVIEW

File: drivers/display/display_ili9340.c [hunk 1]
```diff
@@ -6,9 +6,10 @@
 #include "display_ili9340.h"
 #include <drivers/display/ili9340.h>
 
-#define SYS_LOG_DOMAIN "ILI9340"
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ILI9340_LEVEL
-#include <logging/sys_log.h>
+#define LOG_MODULE_NAME display
+#define LOG_LEVEL CONFIG_LOG_ILI9340_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER();
```
Comment (galak) [inline drivers/display/display_ili9340.c]: This has recently changed, so LOG_MODULE_REGISTER() now takes the LOG_MODULE_NAME, and you don't need to explicit set LOG_MODULE_NAME.  You can just do:

`LOG_MODULE_REGISTER(foo);`

**Review by galak — DISMISSED** [review]
Fix LOG_MODULE_REGISTER for recent changes

File: drivers/display/display_ili9340.c (hunk 1 above)
Comment (vanwinkeljan) [reply drivers/display/display_ili9340.c]: noticed it, fixed

File: samples/display/ili9340/prj.conf [hunk 2]
```diff
@@ -8,16 +8,14 @@ CONFIG_SPI_1=y
...
 CONFIG_ILI9340_CMD_DATA_GPIO_PORT_NAME="GPIOA"
 CONFIG_ILI9340_CMD_DATA_PIN=8
 
-CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_SHOW_TAGS=y
-CONFIG_SYS_LOG_SHOW_COLOR=y
-CONFIG_SYS_LOG_DEFAULT_LEVEL=0
-CONFIG_SYS_LOG_OVERRIDE_LEVEL=0
+CONFIG_LOG=y
+#CONFIG_LOG_SHOW_TAGS=y
```
Comment (nordic-krch) [inline samples/display/ili9340/prj.conf]: do we want to keep those commented lines?

File: samples/display/ili9340/src/main.c [hunk 3]
```diff
@@ -49,15 +50,15 @@ void main(void)
 	dev = device_get_binding("ILI9340");
 
 	if (dev == NULL) {
-		SYS_LOG_ERR("device not found.  Aborting test.");
+		LOG_ERR("device not found.  Aborting test.");
```
Comment (nordic-krch) [inline samples/display/ili9340/src/main.c]: Could you fix log message:
- device->Device 
- one redundant space before 'Aborting'.

**Comment by nashif** [comment]
cherrypicked in #10029

**Comment by vanwinkeljan** [comment]
closing pull request as integrated with #10029

**Pull request closed without merging**